algorithms/kernel/svm/svm_predict_impl.i

```diff
@@ -43,6 +43,134 @@ using namespace daal::internal;
+
+    static Super * create(const size_t nRowsPerBlock, const NumericTablePtr & xTable, const NumericTablePtr & svTable,
+                          kernel_function::KernelIfacePtr & kernel)
+    {
+        auto val = new PredictTaskDense(nRowsPerBlock, xTable, svTable, kernel);
+        if (val && val->isValid()) return val;
+        delete val;
+        return nullptr;
+    }
+
+protected:
+    PredictTaskDense(const size_t nRowsPerBlock, const NumericTablePtr & xTable, const NumericTablePtr & svTable,
+                     kernel_function::KernelIfacePtr & kernel)
+        : Super(nRowsPerBlock, xTable, svTable, kernel)
+    {}
+
+    NumericTablePtr getBlockNTData(const size_t startRow, const size_t nRows, services::Status & status) override
+    {
+        algorithmFPType * xData = const_cast<algorithmFPType *>(_xBlock.set(*Super::_xTable, startRow, nRows));
+        if (!xData) status |= services::Status(services::ErrorMemoryAllocationFailed);
+        return HomogenNumericTableCPU<algorithmFPType, cpu>::create(xData, Super::_nFeatures, nRows, &status);
+    }
+
+private:
+    ReadRows<algorithmFPType, cpu> _xBlock;
+};
+
+template <typename algorithmFPType, CpuType cpu>
+class PredictTaskCSR : PredictTask<algorithmFPType, cpu>
+{
+public:
+    using Super = PredictTask<algorithmFPType, cpu>;
+    DAAL_NEW_DELETE();
+    virtual ~PredictTaskCSR() {}
+
+    static Super * create(const size_t nRowsPerBlock, const NumericTablePtr & xTable, const NumericTablePtr & svTable,
+                          kernel_function::KernelIfacePtr & kernel)
+    {
+        auto val = new PredictTaskCSR(nRowsPerBlock, xTable, svTable, kernel);
+        if (val && val->isValid()) return val;
+        delete val;
+        return nullptr;
+    }
+
+protected:
+    PredictTaskCSR(const size_t nRowsPerBlock, const NumericTablePtr & xTable, const NumericTablePtr & svTable,
+                   kernel_function::KernelIfacePtr & kernel)
+        : Super(nRowsPerBlock, xTable, svTable, kernel), _rowOffsets(nRowsPerBlock + 1)
+    {}
+
+    NumericTablePtr getBlockNTData(const size_t startRow, const size_t nRows, services::Status & status) override
+    {
+        _xBlock.set(dynamic_cast<CSRNumericTableIface *>(Super::_xTable.get()), startRow, nRows);
+        algorithmFPType * values = const_cast<algorithmFPType *>(_xBlock.values());
+        size_t * cols            = const_cast<size_t *>(_xBlock.cols());
+        const size_t * rows      = _xBlock.rows();
+        if (!values || !cols || !rows) status |= services::Status(services::ErrorMemoryAllocationFailed);
+        _rowOffsets[0] = 1;
+        for (size_t i = 0; i < nRows; i++)
+        {
+            const size_t nNonZeroValuesInRow = rows[i + 1] - rows[i];
+            _rowOffsets[i + 1]               = _rowOffsets[i] + nNonZeroValuesInRow;
+        }
+
+        return CSRNumericTable::create(values, cols, _rowOffsets.get(), Super::_nFeatures, nRows, CSRNumericTableIface::CSRIndexing::oneBased,
```

Comment:
potential problem with illegal instruction here. Is it possible to use CPU specific create like for `HomogenNumericTableCPU`?

Reply:
I doubt that the CSR tables use specific instructions. Because in the code we constantly use them inside the library - like here. There have never been problems. If we want to make a specific CPU, then we need to do it within the entire library. Not within the PR.